server/bin/test_data_importer.py

```diff
@@ -0,0 +1,706 @@
+#!/usr/bin/env python
+from __future__ import print_function, division, absolute_import
+
+"""
+Injects synthetic data from one or more JSON files into Candlepin for testing
+"""
+
+import datetime
+import logging
+import json
+import os
+import random
+import re
+import requests
+import string
+import sys
+import time
+
+from optparse import OptionParser
+
+
+
+# Disable HTTP warnings that'll be kicked out by urllib (used internally by requests)
+requests.packages.urllib3.disable_warnings()
+
+# Setup logging
+LOGLVL_TRACE = 5
+logging.addLevelName(LOGLVL_TRACE, 'TRACE')
+logging.basicConfig(stream=sys.stdout, level=logging.INFO, format="%(asctime)-15s %(levelname)-7s %(name)-16s -- %(message)s")
+
+# We don't care about urllib/requests logging
+logging.getLogger("requests").setLevel(logging.WARNING)
+logging.getLogger("urllib3").setLevel(logging.WARNING)
+
+log = logging.getLogger('test_data_importer')
+
+
+# TODO: There's likely a better python way of doing this that I don't know, but all the ways that I've
+# learned make shallow copies of the entire dictionary to do the check which seems rather wasteful to me,
+# so we go old school here.
+def ci_in(dict, search):
+    search = search.lower()
+
+    for key in dict:
+        if key.lower() == search:
+            return True
+
+    return False
+
+# Merges two dictionaries, converting the keys in the source dictionary from snake case to camel case
+# where appropriate.
+def safe_merge(source, dest, include=None, exclude=None):
+    def handle_match(match):
+        return "%s%s" % (match.group(1), match.group(2).upper())
+
+    for key, value in source.items():
+        key = re.sub('(\w)_(\w)', handle_match, key)
+
+        if (include is None or key in include) and (exclude is None or key not in exclude):
+            dest[key] = value
+
+    return dest
+
+def random_id(prefix, suffix_length=8):
+    suffix = ''.join(random.choice(string.digits) for i in range(suffix_length))
+    return "%s-%s" % (prefix, suffix)
+
+
+
+class Candlepin:
+    # Server configuration
+    host = 'localhost'
+    port = 8443
+    username = 'admin'
+    password = 'admin'
+
+    status_cache = None
+    hosted_test = None
+
+
+    def __init__(self, host, port, username, password):
```

Comment:
Please use default values like `host='localhost', port=8443, ...` instead of class attributes.

Reply:
Done